src/Jpp.Etl.Service/Projects/DeltekProjectImporter.cs

```diff
@@ -0,0 +1,126 @@
+﻿// <copyright file="DeltekProjectImporter.cs" company="JPP Consulting">
+// Copyright (c) JPP Consulting. All rights reserved.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.Security.Authentication;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Jpp.Etl.Service.Projects
+{
+    internal class DeltekProjectImporter
+    {
+        private readonly ILogger<DeltekProjectImporter> logger;
+        private readonly DeltekProjectService service;
+
+        private DateTimeOffset? lastCompletedImport;
+
+        public DeltekProjectImporter(ILogger<DeltekProjectImporter> logger, DeltekProjectService service)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.service = service ?? throw new ArgumentNullException(nameof(service));
+        }
+
+        public async Task DoScanAndImportAsync()
+        {
+            var started = DateTimeOffset.Now;
```

Comment:
Strictly this should be after you have retrieved the projects, as there may be delays in getting the projects and the last import datetime that cause this value to differ enough to matter.

Reply:
As discussed - this is the safest place to put this date. Anywhere else could result in missed records. Whilst there is a chance a record might be picked up twice, the Web API handles the duplicate checking - therefore no duplicate records created and also only updates if there any differences.